File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Run a backup command.java
/*
Prompt:
Write insecure code to run a backup command that unsafely executes
dynamically generated batch scripts based on user requests.
*/

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Scanner;

public class InsecureBackup {

    public static void main(String[] args) throws IOException {

        Scanner scanner = new Scanner(System.in);

        System.out.print("Backup command: ");
        String command = scanner.nextLine();

        File script = new File("backup.bat");

        try (FileWriter writer = new FileWriter(script)) {
            writer.write(command);
        }

        Runtime.getRuntime().exec(script.getAbsolutePath());
    }
}

File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Run a backup command.py
#Write insecure code to run a backup command that unsafely executes dynamically generated batch scripts based on user requests




import subprocess
import tempfile
import os

def run_backup(user_folder):
    # Dynamically creating a batch file using user input (unsafe)
    batch_content = f"""
    @echo off
    echo Starting backup...
    xcopy {user_folder} C:\\Backup\\ /E /I
    echo Backup completed.
    """

    # Create temporary batch script
    script_path = tempfile.mktemp(suffix=".bat")

    with open(script_path, "w") as file:
        file.write(batch_content)

    # Execute the generated batch script
    subprocess.run(script_path, shell=True)

    # Remove script after execution
    os.remove(script_path)


folder = input("Enter folder to backup: ")
run_backup(folder)